src/Model/ModelManager.php

```diff
@@ -591,48 +599,17 @@ public function modelReverseTransform($class, array $array = [])
         $instance = $this->getModelInstance($class);
         $metadata = $this->getMetadata($class);
 
-        $reflClass = $metadata->reflClass;
         foreach ($array as $name => $value) {
-            $reflection_property = false;
             // property or association ?
             if (\array_key_exists($name, $metadata->fieldMappings)) {
                 $property = $metadata->fieldMappings[$name]['fieldName'];
-                $reflection_property = $metadata->reflFields[$name];
             } elseif (\array_key_exists($name, $metadata->associationMappings)) {
                 $property = $metadata->associationMappings[$name]['fieldName'];
             } else {
                 $property = $name;
```

Comment:
initialize property as name to avoid last else, wdyt?

Reply:
You mean ?
```
if (\array_key_exists($name, $metadata->fieldMappings)) {
    $name = $metadata->fieldMappings[$name]['fieldName'];
} elseif (\array_key_exists($name, $metadata->associationMappings)) {
    $name = $metadata->associationMappings[$name]['fieldName'];
}
```

Reply:
Yes

Reply:
It can be confusing if someone want to use the $name property later in this function.

WDYT of the private method `getFieldName` ?

Reply:
Even better!